File: app/src/main/java/com/msd/notebook/adapter/InstructorAdapter.java
package com.msd.notebook.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.msd.notebook.R;
import com.msd.notebook.databinding.InstructorItemBinding;
import com.msd.notebook.models.Instructor;

import java.util.ArrayList;
import java.util.List;

public class InstructorAdapter extends RecyclerView.Adapter<InstructorAdapter.ViewHolder> {

    Context context;
    List<Instructor> instructorList = new ArrayList<>();
    InstructorItemClick instructorItemClick;

    public InstructorAdapter(Context context, InstructorItemClick instructorItemClick) {
        this.context = context;
        this.instructorItemClick = instructorItemClick;
    }

    public void setInstructorList(List<Instructor> instructorList) {
        this.instructorList = instructorList;
        notifyDataSetChanged();
    }


    @NonNull
    @Override
    public InstructorAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.instructor_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull InstructorAdapter.ViewHolder holder, int position) {
        holder.binding.instructorNameTxt.setText(instructorList.get(position).getInstructorName());
        holder.binding.instructorId.setText(instructorList.get(position).getId());

        holder.binding.itemRoot.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                instructorItemClick.itemClick(instructorList.get(holder.getAdapterPosition()));
            }
        });

        holder.binding.deleteCard.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                instructorItemClick.itemCardClick(instructorList.get(holder.getAdapterPosition()));
            }
        });
    }

    @Override
    public int getItemCount() {
        return instructorList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        InstructorItemBinding binding;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            binding = InstructorItemBinding.bind(itemView);
        }
    }

    public interface InstructorItemClick {
        void itemClick(Instructor instructor);
        void itemCardClick(Instructor instructor);
    }
}

File: app/src/main/java/com/msd/notebook/adapter/InstructorAdapter.kt
package com.msd.notebook.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.msd.notebook.R
import com.msd.notebook.databinding.InstructorItemBinding
import com.msd.notebook.models.Instructor

class InstructorAdapter(var context: Context, var instructorItemClick: InstructorItemClick) :
    RecyclerView.Adapter<InstructorAdapter.ViewHolder>() {
    var instructorList : ArrayList<Instructor>? = null
    fun updateInstructorList(instructorList: ArrayList<Instructor>) {
        this.instructorList = instructorList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.instructor_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.binding.instructorNameTxt.text = instructorList?.get(position)!!.instructorName
        holder.binding.instructorId.text = instructorList!![position].id
        holder.binding.itemRoot.setOnClickListener {
            instructorItemClick.itemClick(
                instructorList!![holder.getAdapterPosition()]
            )
        }
        holder.binding.deleteCard.setOnClickListener {
            instructorItemClick.itemCardClick(
                instructorList!![holder.getAdapterPosition()]
            )
        }
    }

    override fun getItemCount(): Int {
        return instructorList!!.size
    }

    interface InstructorItemClick {
        fun itemClick(instructor: Instructor?)
        fun itemCardClick(instructor: Instructor?)
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var binding: InstructorItemBinding

        init {
            binding = InstructorItemBinding.bind(itemView)
        }
    }
}
